plugin/src/test/java/org/opensearch/ml/action/connector/GetConnectorTransportActionTests.java

```diff
@@ -87,82 +120,103 @@ public void setup() throws IOException {
         threadContext = new ThreadContext(settings);
         when(client.threadPool()).thenReturn(threadPool);
         when(threadPool.getThreadContext()).thenReturn(threadContext);
+        when(threadPool.executor(anyString())).thenReturn(testThreadPool.executor(GENERAL_THREAD_POOL));
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        ThreadPool.terminate(testThreadPool, 500, TimeUnit.MILLISECONDS);
     }
 
-    public void testGetConnector_UserHasNodeAccess() throws IOException {
+    public void testGetConnector_UserHasNodeAccess() throws IOException, InterruptedException {
         doAnswer(invocation -> {
             ActionListener<Boolean> listener = invocation.getArgument(3);
             listener.onResponse(false);
             return null;
         }).when(connectorAccessControlHelper).validateConnectorAccess(any(), any(), any());
 
         GetResponse getResponse = prepareConnector();
-        doAnswer(invocation -> {
-            ActionListener<GetResponse> listener = invocation.getArgument(1);
-            listener.onResponse(getResponse);
-            return null;
-        }).when(client).get(any(), any());
+        PlainActionFuture<GetResponse> future = PlainActionFuture.newFuture();
+        future.onResponse(getResponse);
+        when(client.get(any(GetRequest.class))).thenReturn(future);
```

Comment:
Why `client.get`? Shouldn't we use `sdkClient`? Are you sure these tests are reflecting the actual behavior we have in the class?

Reply:
Thanks for the explanation.